File: frontend/src/types/emailTemplate.ts
/**
 * Email template types for COA email configuration.
 */

/** A single template variable definition */
export interface EmailTemplateVariable {
  key: string
  description: string
}

/** Response containing available template variables */
export interface EmailTemplateVariablesResponse {
  variables: EmailTemplateVariable[]
}

/** Email template data for updates */
export interface EmailTemplateUpdate {
  subject: string
  body: string
}

/** Email template response from API */
export interface EmailTemplate {
  id: number
  name: string
  subject: string
  body: string
  created_at: string
  updated_at: string | null
}

/** Preview of rendered email template */
export interface EmailTemplatePreview {
  subject: string
  body: string
}

File: backend/app/schemas/email_template.py
"""Email template schemas for request/response validation."""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class EmailTemplateVariable(BaseModel):
    """Schema for a single template variable."""

    key: str
    description: str


class EmailTemplateVariablesResponse(BaseModel):
    """Response schema for available template variables."""

    variables: List[EmailTemplateVariable]


class EmailTemplateBase(BaseModel):
    """Base email template schema."""

    subject: str = Field(..., min_length=1, max_length=500)
    body: str = Field(..., min_length=1)


class EmailTemplateUpdate(EmailTemplateBase):
    """Schema for updating an email template."""

    pass


class EmailTemplateResponse(BaseModel):
    """Email template response schema."""

    id: int
    name: str
    subject: str
    body: str
    created_at: datetime
    updated_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class EmailTemplatePreview(BaseModel):
    """Preview of rendered email template."""

    subject: str
    body: str
